atom/common/native_mate_converters/v8_value_converter.cc

```diff
@@ -243,12 +247,21 @@ base::Value* V8ValueConverter::FromV8ValueImpl(
     return NULL;
 
   if (val->IsDate()) {
-    if (!date_allowed_)
-      // JSON.stringify would convert this to a string, but an object is more
-      // consistent within this class.
-      return FromV8Object(val->ToObject(), state, isolate);
     v8::Date* date = v8::Date::Cast(*val);
-    return new base::FundamentalValue(date->NumberValue() / 1000.0);
+    double const time_ms = date->NumberValue();
+    int const time_in_day_ms = TimeInDay(time_ms);
+    std::time_t time_s(time_ms / 1000.0);
+    char buf[128];
+    std::tm* gm_time = std::gmtime(&time_s);
+    int year = gm_time->tm_year + 1900;
+    int month  = gm_time->tm_mon + 1;
+    int ms = time_in_day_ms % 1000;
+    size_t length = snprintf(buf, sizeof buf,
+                             "%04d-%02d-%02dT%02d:%02d:%02d.%03dZ",
```

Comment:
It is probably better to call the `toISOString` method directly since it is not easy to get this right:
https://code.google.com/p/chromium/codesearch#chromium/src/v8/src/builtins.cc&sq=package:chromium&type=cs&l=3179